### ISSUE TITLE
DownwardAPI fails validation in CRD

### ISSUE THREAD

**Issue by cliveseldon** [SeldonIO/seldon-core#926]
The latest CRD has full OpenAPI validation auto-generated by Kubebuilder. However, the PodSpec validation has issues with downwardAPI which will cause failures of the form:

```
type in spec.predictors.componentSpecs.spec.volumes.downwardAPI is required
```
This is caused by: https://github.com/go-openapi/validate/issues/108
A fix has been created: https://github.com/go-openapi/validate/pull/107 but does not seem to be in a k8s release yet.

A temporary solution may be needed. 

This will become a greater issue if not fully fixed when k8s only allows structural schema as then all valid fields must be defined in the OpenAPI validationspec.

**Comment by cliveseldon** [comment]
The core library that needs to use the fix seems to be: https://github.com/kubernetes/apiextensions-apiserver/issues/42

**Comment by cliveseldon** [comment]
In theory the fix is available in k8s 0.16.1 but not been able to test in Kind yet.

**Comment by adriangonz** [comment]
Just to add to this, I've also seen something similar when trying to mount a `configMap` as a `volume`. In which case the error is `type in spec.predictors.componentSpecs.spec.volumes.configMap is required`. 

This error can be replicated using [this notebook](https://github.com/SeldonIO/seldon-core/blob/master/examples/models/template_model_tracing/tracing.ipynb). 

I'm assuming the fix for both is probably similar?

**Comment by VishDev12** [comment]
Ran into this issue in the example mentioned by @adriangonz . Is there a temporary fix for this?

**Comment by cliveseldon** [comment]
I opened a bug on k8s: https://github.com/kubernetes/kubernetes/issues/84880

**Comment by cliveseldon** [comment]
Need to see if we can remove the OpenAPISchema for these and allow unknownFields

**Comment by cliveseldon** [comment]
Temporary fix in #1069 @VishDev12

**Comment by VishDev12** [comment]
Will check @cliveseldon Thank you

**Comment by cliveseldon** [comment]
Upstream Kubernetes will be fixed in  v1.18, v1.17
